### PR TITLE
Fix panic in optimizer translater

### PR DESCRIPTION
**Pull request by xiong-gang**
CTranslatorRelcacheToDXL::RetrieveIndex() access 'rd_index' after the relation
is closed, it would panic when compiled with '-DRELCACHE_FORCE_RELEASE' flag.

Co-authored-by: Alexandra Wang <lewang@pivotal.io>
Co-authored-by: Gang Xiong <gxiong@pivotal.io>

### CODE REVIEW

**Comment by xiong-gang** [comment]
Should backport to 5X

File: src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp [hunk 1]
```diff
@@ -1086,6 +1089,7 @@ CTranslatorRelcacheToDXL::RetrieveIndex
 		}
 
 		form_pg_index = index_rel->rd_index;
+		index_clustered = form_pg_index->indisclustered;
 		GPOS_ASSERT (NULL != form_pg_index);
```
Comment (dhanashreek89) [inline src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp]: Lets move this after the ASSERT

Comment (xiong-gang) [reply src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp]: right, thanks.

**Comment by dhanashreek89** [comment]
LGTM. Thanks!
Did this surface because of an existing test? if no, would it be possible to add a test for this?

**Comment by xiong-gang** [comment]
@dhanashreek89 
It crashes when I run ICW. So I guess no extra tests are needed here.

Btw, I run ICW again with optimizer on and `DRELCACHE_FORCE_RELEASE` set, find another 2 problems have the same cause. Fixed it in another commit. Please help review it. Thanks.

**Comment by xiong-gang** [comment]
Merged, thanks for the review.

**Pull request merged**